### ISSUE TITLE
Add missing `use_exec` to serve the API

### ISSUE THREAD

**Issue by lsorber** [superlinear-ai/substrate#203]
We should be exec'ing that process, but it seems that Poe the Poet does not support this for `shell` tasks: https://poethepoet.natn.io/tasks/task_types/shell.html#available-task-options

Looks like this can be solved with https://poethepoet.natn.io/tasks/task_types/switch.html.

**Comment by SiPaRadix** [comment]
From the documentation in the link above:

> switch tasks support all of the [standard task options](https://poethepoet.natn.io/tasks/options.html) with the exception of use_exec.

Does that not mean a subprocess is spawned regardless?

Alternatively, could we consider adding `exec` in front of `uvicorn ...` and `gunicorn ...` and avoid spawning a subprocess that way or is it already too late since a subprocess of the poe cli is already spawned at that point?
From the [PyPI project description](https://pypi.org/project/poethepoet/0.18.1/):

> Normally tasks are executed as subprocesses of the poe cli.

**Comment by lsorber** [comment]
Yes, Poe will spawn a subprocess. [Poe added the ability to exec a command at our request a while back](https://github.com/nat-n/poethepoet/issues/47). [And we want to exec the process because of the reasons explained in this comment](https://github.com/nat-n/poethepoet/issues/47#issuecomment-1003370597).

Unfortunately, I don't think adding `exec` will help because that will just replace the process that Poe starts to run the shell script (and not the Poe process itself).

One solution is to replace the current shell switch to a Poe switch task, in which case we can exec the switched commands.
